pages/configuration/high-availability.mdx

```diff
@@ -0,0 +1,264 @@
+Other option is to drop data from instance which is more up to date, by shutting it down and deleting snapshots and WAL directories in 
+`--data-directory`.
+
+
+MAIN
+
+3. Failover logic:
+- single-coordinator
+- whole logic, callbackovi kako to funkcionira
+
+
+- diverged from main -> user has to manually solve it if happens
+- idempotentno demotanje u repliku 
+
+
+- REPLICA can talk only with a single main, possible that cannot talk with any instance for a short time.. -> to avoid inconsitencies :checmark:
+- choosing REPLICA currently simple :Checkmark: 
+
+
+## Instances' restart
+
+
+In 2.15 release, we assume there is a single coordinator instance which cannot fail. Replication instance can fail both as MAIN and as REPLICA.
+When instance which was REPLICA comes back, it won't accept updates from any instance until coordinator updates its responsible peer. This should
+happen automatically when coordinator's ping to the instance passes. When MAIN instance comes back, any writing to the MAIN instance will be forbidden 
+until ping from coordinator passes. When the coordinator realizes that the once-MAIN instance is alive, it will choose between enabling writing on old MAIN
+or demoting it to REPLICA. The choice depends on whether there was any other MAIN instance chosen between old MAIN's failure and restart.
+
+Since instance can die and get back up, use `--replication-state-at-startup=true` to restore last replication state. With this config we will restore instance in 
+state it was last and everything should function normally from that point. In case this flag is not set, every instance will restart as MAIN and that case is not handled.
+
+## Handling errors
+
+Distributed system can fail in various ways. The logic is implemented in a way that Memgraph instance will be resilient to occasional network
+failures and independent machine failures. That's why there are parameters controlling frequency of health checks from coordinator to replication
+instances and the time needed to realize the instance is down.
+
+## Example
+
+This example will show how to setup highly-available cluster in Memgraph using a single coordinator.
+
+### 1. Start all instances
+
+i.) Start coordinator:
+```plaintext
+docker run  --name coord1 --rm -p 7687:7687 -p 7444:7444 memgraph/memgraph-mage --bolt-port=7687 --log-level=TRACE --data-directory=/tmp/mg_data_coord1 --log-file=/tmp/coord1.log --also-log-to-stderr --raft-server-id=1 --raft-server-port=10111 --experimental-enabled=high-availability
+```
+
+ii.) Start instance1:
+```plaintext
+docker run  --name instance1 --rm -p 7688:7688 -p 7445:7444 memgraph/memgraph-mage --bolt-port=7688 --log-level=TRACE --data-directory=/tmp/mg_data_instance1 --log-file=/tmp/instance1.log --also-log-to-stderr --replication-restore-state-on-startup --coordinator-server-port=10011 --experimental-enabled=high-availability
```

Comment:
Consider removing --rm since it will be automatically removed if something fails, hence you need to do config again.

Reply:
Makes sense